components/eamxx/cime_config/eamxx_buildnml_impl.py

```diff
@@ -246,13 +246,22 @@ def refine_type(entry, force_type=None):
             expect (force_type is None or is_array_type(force_type),
                     "Error! Invalid type '{}' for an array.".format(force_type))
 
-            elem_type = force_type if force_type is None else array_elem_type(force_type)
-            result = [refine_type(item.strip(), force_type=elem_type) for item in entry.split(",") if item.strip() != ""]
-            result = make_array(result,'string' if elem_type is None else elem_type)
+            if force_type is None:
+                # Try to derive type from first element
+                elem_type = derive_type([item.strip() for item in entry.split(",")][0])
+            else:
+                elem_type = force_type if force_type is None else array_elem_type(force_type)
```

Comment:
Since you're in the `else` branch, you already know force_type is not None.

Reply:
Good catch, thx